### PR TITLE
bigtable: support custom connector

### PR DESCRIPTION
**Pull request by fanatid**
#### Problem

#25918 add custom connector with `hyper-proxy` but more flexible solution pass custom connector through config. Maybe we will move `BIGTABLE_PROXY` outside of `BigTableConnection::new`? What do you think @lwus @CriesofCarrots?

#### Summary of Changes

Add connector to `LedgerStorageConfig`.

### CODE REVIEW

**Comment by lwus** [comment]
#25918 targeted a specific use case we had. If the generic case is useful, why not completely remove the baked in proxy / http / etc and just expose some helper functions to recreate those?

**Comment by fanatid** [comment]
@lwus I moved `hyper_proxy` to `LedgerStorageConfig::with_proxy_connector`

File: storage-bigtable/src/lib.rs [hunk 1]
```diff
@@ -377,26 +381,56 @@ pub enum CredentialType {
...
+        let proxy = hyper_proxy::Proxy::new(
+            hyper_proxy::Intercept::All,
+            proxy_uri
+                .parse::<http::Uri>()
+                .map_err(|err| Error::InvalidUri(proxy_uri, err.to_string()))?,
+        );
+
+        let mut http = HttpConnector::new();
+        http.enforce_http(false);
+        http.set_nodelay(true);
```
Comment (CriesofCarrots) [inline storage-bigtable/src/lib.rs:423]: Any particular reason for this addition?

**Review by CriesofCarrots — COMMENTED** [review]
I've no strong objection to this, but your PR description doesn't include a problem statement, so I'm not totally sure what need you're trying to meet...?

File: storage-bigtable/src/lib.rs (hunk 1 above)
Comment (fanatid) [reply storage-bigtable/src/lib.rs:423]: I actually described this in my other PR: https://github.com/solana-labs/solana/pull/26398
In `tonic` which is used here in the inner connector `nodelay` is set to true.

**Comment by fanatid** [comment]
> I've no strong objection to this, but your PR description doesn't include a problem statement, so I'm not totally sure what need you're trying to meet...?

Yeah, I did not include it in the PR description :disappointed: I use a custom connector for counting approximate inbound/outbound traffic to bigtable, so it would be possible to see how much we will need to pay.

File: storage-bigtable/src/lib.rs (hunk 1 above)
Comment (CriesofCarrots) [reply storage-bigtable/src/lib.rs:423]: Ah, missed that one. Thanks for the link

**Review by CriesofCarrots — APPROVED** [review]
Okay, lgtm.
@lwus , before I merge, can you confirm that adopting to this new api will not be too disruptive to the use-case you were targeting on upgrade? I can't recall if that was using BigTableConnection directly, or LedgerStorage

**Comment by CriesofCarrots** [comment]
@fanatid , is this still desired?

**Comment by fanatid** [comment]
We still wrap the connector for counting traffic to BigTable but I think this PR is outdated for us, easier would be to close it.

**Pull request closed without merging**